core/src/test/java/org/apache/iceberg/rest/requests/TestCreateNamespaceRequest.java

```diff
@@ -0,0 +1,105 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.util.JsonUtil;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestCreateNamespaceRequest {
+
+  private static final ObjectMapper MAPPER = JsonUtil.withRESTComponents();
+
+  // Non-empty
+  private static final String JSON_REQUEST_STRING =
+      "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{\"owner\":\"Hank\"}}";
+  private static final Namespace NAMESPACE = Namespace.of("accounting", "tax");
+  private static final Map<String, String> PROPERTIES = ImmutableMap.of("owner", "Hank");
+
+  // Empty - The response requires a non-empty namespace
+  private static final String JSON_REQUEST_NO_PROPERTIES =
+      "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{}}";
+
+  @Test
+  public void testDeserialize() throws JsonProcessingException {
+    CreateNamespaceRequest expected = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .withProperties(PROPERTIES)
+        .build();
+
+    CreateNamespaceRequest actual = MAPPER.readValue(JSON_REQUEST_STRING, CreateNamespaceRequest.class);
+
+    assertSame(expected, actual);
+  }
+
+  @Test
+  public void testDeserializeNoProperties() throws JsonProcessingException {
+    CreateNamespaceRequest expected = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .build();
+
+    CreateNamespaceRequest actual = MAPPER.readValue(JSON_REQUEST_NO_PROPERTIES, CreateNamespaceRequest.class);
+
+    assertSame(expected, actual);
+  }
+
+  @Test
+  public void testSerialize() throws JsonProcessingException {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .withProperties(PROPERTIES)
+        .build();
+
+    Assertions.assertThat(MAPPER.writeValueAsString(req))
+        .isEqualTo(JSON_REQUEST_STRING);
+  }
+
+  @Test
+  public void testSerializeNoProperties() throws JsonProcessingException {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .build();
+
+    Assertions.assertThat(MAPPER.writeValueAsString(req))
+        .isEqualTo(JSON_REQUEST_NO_PROPERTIES);
+  }
+
+  private static void assertSame(CreateNamespaceRequest expected, CreateNamespaceRequest actual) {
```

Comment:
I think this should be `assertEquals` rather than `assertSame`. `Assert.assertSame` in the JUnit API tests object identity, not equality and I think we should use the same definition.

Reply:
As per a comment from @nastra, I chose to implement `equals` directly on the class and then this method became unnecessary as I can just do the equality check in line in the test.